beacon_node/lighthouse_network/src/rpc/handler.rs

```diff
@@ -349,21 +349,8 @@ where
     }
 
     fn connection_keep_alive(&self) -> bool {
-        // Check that we don't have outbound items pending for dialing, nor dialing, nor
-        // established. Also check that there are no established inbound substreams.
-        // Errors and events need to be reported back, so check those too.
         match self.state {
-            HandlerState::ShuttingDown(_) => {
-                !self.dial_queue.is_empty()
-                    || !self.outbound_substreams.is_empty()
-                    || !self.inbound_substreams.is_empty()
-                    || !self.events_out.is_empty()
-                    || !self.dial_negotiated != 0
```

Comment:
this feels redundant, the `KeepAlive` algorithm [already covers some of these cases implicitly](https://docs.rs/libp2p/latest/libp2p/swarm/trait.ConnectionHandler.html#keep-alive-algorithm), 
and the state where the handler is shutting down with all those properties toggled is covered when `poll`ing which also changes the connection handler state to `Deactivated` which we cannot do here.

Reply:
Yeah, this doesn't really matter. We never really rely on the keep-alive, because it is combined with all the other behaviours and the gossipsub keep-alive i think maintains most connections